### ISSUE TITLE
Allow downloading webpages with `--write-pages` `--skip-download`

### ISSUE THREAD

**Issue by hacker-DOM** [yt-dlp/yt-dlp#10102]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a feature unrelated to a specific site
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

Tldr: In the case of `yt-dlp --write-pages --skip-download {url}`, don't look for embeds (or allow to ignore them silently with `-i` (--ignore-errors).

I have a download pipeline where I first download the htmlpage, look for links there, then download these links. "`yt-dlp` is not made for downloading htmlpages!!" you might say. True, but it provides the best user experience (wget, curl, httpie etc. don't allow you to reuse Chrome cookies with one flagoption; in fact I don't know any other tool that does).

This pipeline worked great until I encountered a page that didn't have a video on it. That is when I discovered `yt-dlp` throws--I believe this to be unexpected, and undesired.
Thx!

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
❯ yt-dlp --write-pages --skip-download 'https://grapplersguide.com/second-portal/lms/courses/fundamental-positional-techniques.31689/' --cookies-from-browser chrome -vU
[debug] Command-line config: ['--write-pages', '--skip-download', 'https://grapplersguide.com/second-portal/lms/courses/fundamental-positional-techniques.31689/', '--cookies-from-browser', 'chrome', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.05.27 from yt-dlp/yt-dlp [12b248ce6] (pip)
[debug] Python 3.11.5 (CPython arm64 64bit) - macOS-14.0-arm64-arm-64bit (OpenSSL 3.3.0 9 Apr 2024)
[debug] exe versions: ffmpeg 7.0 (setts), ffprobe 7.0
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.02.02, mutagen-1.47.0, requests-2.31.0, sqlite3-3.45.3, urllib3-2.2.0, websockets-12.0
[debug] Proxy map: {}
Extracting cookies from chrome
[debug] Extracting cookies from: "/Users/dteiml/Library/Application Support/Google/Chrome/Default/Cookies"
[debug] using find-generic-password to obtain password from OSX keychain
Extracted 1161 cookies from chrome
[debug] cookie version breakdown: {'v10': 1179, 'other': 0, 'unencrypted': 9}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1820 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.05.27 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.05.27 from yt-dlp/yt-dlp)
[generic] Extracting URL: https://grapplersguide.com/second-portal/lms/courses/fundamental-positional-techniques.31689/
[generic] fundamental-positional-techniques: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] Saving request to fundamental-positional-techniques_https_-_grapplersguide.com_second-portal_lms_courses_fundamental-positional-techniques.31689_.dump
[generic] fundamental-positional-techniques: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://grapplersguide.com/second-portal/lms/courses/fundamental-positional-techniques.31689/
Traceback (most recent call last):
  File "/Users/dteiml/.local/pipx/venvs/yt-dlp/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 1606, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dteiml/.local/pipx/venvs/yt-dlp/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 1741, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "/Users/dteiml/.local/pipx/venvs/yt-dlp/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 734, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dteiml/.local/pipx/venvs/yt-dlp/lib/python3.11/site-packages/yt_dlp/extractor/generic.py", line 2514, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://grapplersguide.com/second-portal/lms/courses/fundamental-positional-techniques.31689/
```

**Comment by bashonly** [comment]
`--write-pages` is a developer option intended for use with debugging/writing extractors. 

> "yt-dlp is not made for downloading htmlpages!!" you might say.

yt-dlp is not made for downloading htmlpages!!

Even if this wasn't out-of-scope, though, I'm not exactly sure what you want yt-dlp to do in this case. Just exit w/o error code? The html is still being dumped:

```
[generic] Saving request to fundamental-positional-techniques_https_-_grapplersguide.com_second-portal_lms_courses_fundamental-positional-techniques.31689_.dump
```

You can just ignore the error code in your script, no?

> wget, curl, httpie etc. don't allow you to reuse Chrome cookies with one flagoption; in fact I don't know any other tool that does

What you could do is only use yt-dlp to extract the cookies from browser and write them to a cookiefile,. e.g. `yt-dlp --cookies-from-browser firefox --cookies file_to_write_the_cookies_to.txt`, and then use curl/etc for the rest
